dev_scripts_helpers/github/sync_gh_issue_labels.py

```diff
@@ -0,0 +1,326 @@
+
+        :param owner: GitHub repository owner/organization
+        :param repo: GitHub repository name
+        :param name: label name to delete
+        """
+        url = f"https://api.github.com/repos/{owner}/{repo}/labels/{name}"
+        response = requests.delete(url, headers=self.headers)
+        response.raise_for_status()
+        _LOG.info("Label deleted: %s", name)
+
+
+def load_labels(path: str) -> List[Label]:
+    """
+    Load labels from label inventory manifest file.
+
+    :param path: path to label inventory manifest file
+    :return: label objects
+    """
+    try:
+        with open(path, "r") as file:
+            yaml_data = yaml.safe_load(file)
+            labels = [
+                Label(
+                    name=item["name"],
+                    description=item["description"],
+                    color=item["color"],
+                )
+                for item in yaml_data
+            ]
+            return labels
+    except Exception as e:
+        _LOG.error("Error loading label inventory manifest file: %s", str(e))
+        sys.exit(1)
+
+
+def save_labels(labels: List[Label], path: str) -> None:
+    """
+    Save labels to the label inventory manifest file.
+
+    :param labels: label objects
+    :param path: path to save the label inventory manifest file to
+    """
+    try:
+        with open(path, "w") as file:
+            yaml_data = [
+                {
+                    "name": label.name,
+                    "description": label.description,
+                    "color": f"#{label.color}",
+                }
+                for label in labels
+            ]
+            yaml.dump(yaml_data, file, default_flow_style=False, sort_keys=False)
+    except Exception as e:
+        _LOG.error("Error saving label inventory manifest file: %s", str(e))
+        sys.exit(1)
+
+
+def sync_labels(
+    client: GitHubClient, owner: str, repo: str, labels: List[Label], prune: bool
+) -> None:
+    """
+    Synchronize labels between config and repository.
+
+    :param client: GitHub client
+    :param owner: GitHub repository owner/organization
+    :param repo: GitHub repository name
+    :param labels: label objects
+    :param prune: delete labels that exist in the repo but not in the
+        label inventory manifest file
+    """
+    # Build maps for efficient lookup.
+    label_map = {label.name: label for label in labels}
+    current_labels = client.get_labels(owner, repo)
+    current_label_map = {label.name: label for label in current_labels}
+    # Delete labels if pruning is enabled.
+    if prune:
+        for current_label in current_labels:
+            if current_label.name not in label_map:
+                try:
+                    client.delete_label(owner, repo, current_label.name)
+                except Exception as e:
+                    _LOG.error(
+                        "Error deleting label %s: %s", current_label.name, e
+                    )
```

Comment:
My (or ours) style is for objects to throw exceptions (typically dasserts) that have meaningful messages.
In this case client.delete_label does the work and if it can't, dasserts and the whole thing stops.

Catching and rethrowing is useful only if you can actually do something (rarely possible!) or you can add more context to the error.
In this case neither is possible so we should not catch and rethrow.